src/Send/Context/SendContext.tsx

```diff
@@ -14,58 +14,41 @@ export const SendProvider: React.FC<SendProvider> = ({children, wallet}) => {
 
   const [state, dispatch] = React.useReducer(sendReducer, {
     ...initialState,
-    primaryTokenId,
     selectedTokenId: primaryTokenId,
   })
 
   return (
     <SendContext.Provider
       value={{
         ...state,
-        changeSelectedTokenId: (selectedTokenId) =>
+        receiverChanged: (receiver) =>
           dispatch({
-            type: 'changeSelectedTokenId',
-            payload: {
-              selectedTokenId,
-            },
+            type: 'receiverChanged',
+            receiver,
           }),
-        changeReceiver: (receiver) =>
+        amountChanged: (amount) =>
           dispatch({
-            type: 'changeReceiver',
-            payload: {
-              receiver,
-            },
+            type: 'amountChanged',
+            amount,
           }),
-        changeAmount: (amount) =>
+        sendAllCheckboxSelected: () =>
```

Comment:
ui should drive the events, but the reducer should not be coupled to the ui implementation

Reply:
Done